Chap_API_Proc_Mgmt.tex

```diff
@@ -239,7 +217,7 @@ \subsection{\code{PMIx_Spawn_nb}}
 \reqattrend
 
 \optattrstart
-The following attributes are optional for host environments that support this operation:
+The following attributes are optional for \ac{PMIx} implementations:
```

Comment:
Hmmm...those operations are not performed within PMIx. The attribute is simply communicated to the host for execution. I suspect you are trying to say that a PMIx implementation doesn't have to define them, but that doesn't make much sense as we don't provide that option elsewhere. You have to define it - the host just may not implement support for it.

Reply:
So we generally have a mix of the following:

Reply:
This chapter even has sections where some attributes are specifically called out as optional for PMIx implementations and other attributes are marked as optional for host environments for the same API.

Reply:
Sure - it is optional to implement support for those attributes. We require support for others, and we are very clear which ones are optional for the implementation vs the host vs both.

It is critical that PMIx remain in the "messenger" role and not shift into being the "doer" of things. We do internally support PMIx functions, but we want to be very careful and clear where the line lies. And no, that isn't optional across PMIx implementations - or else the host now has to deal with uncertainty over what the implementation may or may not do, which will be very unpopular (should there ever be another implementation).